File: src/PlayQueensGameLinks.php
<?php

declare(strict_types=1);

namespace PlayQueensGame\SiteKit;

final class PlayQueensGameLinks
{
    public const BASE_URL = 'https://playqueensgame.org';

    public static function homeUrl(): string
    {
        return self::BASE_URL;
    }

    public static function pageUrl(string $slug): string
    {
        $clean = trim($slug, '/');
        return $clean === '' ? self::BASE_URL : self::BASE_URL . '/' . $clean . '/';
    }

    public static function playUrl(): string
    {
        return self::BASE_URL . '/#play';
    }

    public static function dailyUrl(): string
    {
        return self::pageUrl('daily');
    }

    public static function learnUrl(): string
    {
        return self::pageUrl('learn');
    }

    public static function blogUrl(): string
    {
        return self::pageUrl('blog');
    }
}

File: src/main/java/org/playqueensgame/sitekit/PlayQueensGameLinks.java
package org.playqueensgame.sitekit;

/**
 * URL helpers for Queens Game.
 */
public final class PlayQueensGameLinks {
    /** Queens Game homepage. */
    public static final String BASE_URL = "https://playqueensgame.org";

    private PlayQueensGameLinks() {
    }

    /**
     * Returns the Queens Game homepage URL.
     *
     * @return homepage URL
     */
    public static String homeUrl() {
        return BASE_URL;
    }

    /**
     * Builds an Queens Game page URL from a path slug.
     *
     * @param slug page path slug
     * @return page URL
     */
    public static String pageUrl(String slug) {
        String clean = slug == null ? "" : slug.replaceAll("^/+|/+$", "");
        return clean.isEmpty() ? BASE_URL : BASE_URL + "/" + clean + "/";
    }

    /**
     * Returns the Queens Game play board URL.
     *
     * @return play board URL
     */
    public static String playUrl() {
        return BASE_URL + "/#play";
    }

    /**
     * Returns the Daily Queens URL.
     *
     * @return Daily Queens URL
     */
    public static String dailyUrl() {
        return pageUrl("daily");
    }

    /**
     * Returns the Queens Game learn URL.
     *
     * @return learn URL
     */
    public static String learnUrl() {
        return pageUrl("learn");
    }

    /**
     * Returns the Queens Game blog URL.
     *
     * @return blog URL
     */
    public static String blogUrl() {
        return pageUrl("blog");
    }
}
